### PR TITLE
Updating for GitHub Actions

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR addresses #27.

### CODE REVIEW

**Comment by jGaboardi** [comment]
Also addressing #14

**Comment by knaaptime** [comment]
this is awesome. I also have some time set aside this week to look at https://github.com/pypa/gh-action-pypi-publish and https://github.com/actions/create-release. will report back with findings

File: setup.py [hunk 1]
```diff
@@ -1,60 +1,67 @@
...
+        version=__version__,
+        description="",  # short <80chr description
+        url="",  # github repo
+        maintainer="",
+        maintainer_email="",
+        setup_requires=["pytest-runner"],
+        tests_require=["pytest"],
+        keywords="spatial statistics",
+        classifiers=[
+            "Development Status :: 5 - Production/Stable",
```
Comment (knaaptime) [inline setup.py:45]: should we maybe default to beta for new packages?

Comment (sjsrey) [reply setup.py:45]: or even alpha?

Comment (jGaboardi) [reply setup.py:45]: we like `Development Status :: 3 - Alpha` or `Development Status :: 4 - Beta` better?

Comment (jGaboardi) [reply setup.py:45]: I'll make the change to this PR once we decide.

Comment (sjsrey) [reply setup.py:45]: alpha +1

**Comment by jGaboardi** [comment]
@knaaptime FYI, `rtree` is working on a [GHA publishing solution](https://github.com/Toblerity/rtree/blob/7665dd4501da80642a2c5955cc12777d8434aef6/.github/workflows/build.yml#L211), also.

File: setup.py [hunk 2]
```diff
@@ -1,60 +1,67 @@
...
-if __name__ == '__main__':
+    setup(
+        name="PACKAGE_NAME",  # name of package
+        version=__version__,
+        description="",  # short <80chr description
+        url="",  # github repo
+        maintainer="",
+        maintainer_email="",
+        setup_requires=["pytest-runner"],
+        tests_require=["pytest"],
```
Comment (weikang9009) [inline setup.py:42]: @jGaboardi Does transitioning from `nosetests` to `pytest` require the submodule to rewrite the tests suite if it is still using unittest?

Comment (jGaboardi) [reply setup.py:42]: @weikang9009 I did not have to change anything within the `test_*.py` files themselves. The only changes are those there in lines 41 and 42, and also [these here](https://github.com/pysal/spaghetti/blob/656c0b65304c7d026c1be708c09bd7da9d584423/setup.py#L7-L11) related to `conda-forge` releases (I did not add them to the PR to keep in bare bones).

**Pull request merged**